changes/ee/feat-13452.en.md

```diff
@@ -0,0 +1 @@
+Added to possibility to configure a list of predefined Kafka topics to Kafka producer actions, and also to use templates to define the destination Kafka topic.
```

Comment:
Please verify if the impact of unknown topic is correct below.
```suggestion
Kafka producer's `topic` config now supports template.
The topics must be created already in Kafka. If a message is rendered towards a non-existing topic in Kafka (given Kafka disabled topic auto-creation), the message will fail with an unrecoverable error. This can cause other messages in the same action to delay.
```

Reply:
Yes, it'll result in an unrecoverable error.  We check for that in this test line: https://github.com/emqx/emqx/pull/13518/files#diff-a270708602dcaf19e272a18dd76d32b8378d5951b49fad623ea9d975889a01e1R1036

The only small thing I'd say is that I _think_ the "this can cause other messages in the same action to delay" won't apply to Kafka, since it doesn't use builtin buffer workers that may block.

```suggestion
Kafka producer action's `topic` config now supports templates.

The topics must be already created in Kafka. If a message is rendered towards a non-existing topic in Kafka (given Kafka disabled topic auto-creation), the message will fail with an unrecoverable error.  Also, if a message does not contain enough information to render to the configured template (e.g.: the template is `t-${t}` and the message context does not define `t`), this message will also fail with an unrecoverable error. 

This same feature is also available for Azure Event Hubs and Confluent Platform producer integrations.
```

Reply:
> Yes, it'll result in an unrecoverable error. We check for that in this test line: https://github.com/emqx/emqx/pull/13518/files#diff-a270708602dcaf19e272a18dd76d32b8378d5951b49fad623ea9d975889a01e1R1036
> 
> The only small thing I'd say is that I _think_ the "this can cause other messages in the same action to delay" won't apply to Kafka, since it doesn't use builtin buffer workers that may block.

but unrecoverable error would result in resource state transition which needs time to recover from health checks ?